### PR TITLE
#137090993/jasmine karma test for front end

### PR DESCRIPTION
**Pull request by cnwosu**
#### What does this PR do?
 Test the front end
#### Description of Task to be completed? 
To ensure the front end has all functionality work the way it was intended. We used Jasmine and Karma(with PhantomJS, Chrome and Firefox) to setup front end testing environment.

#### How should this be manually tested? 
run 'npm install' to install all dependencies in the package.json

### CODE REVIEW

**Review by nyambati — CHANGES_REQUESTED** [review]
The are a lot of HoundCI Issues and a merge conflict in your PR. Resolve this issues first.

You can resolve the merge conflicts by updating your branch with the latest code from `develop` branch.

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    /* frameworks to use
+     available frameworks: 
+     https://npmjs.org/browse/keyword/karma-adapter */
+    
+    frameworks: ['jasmine'],
+
+
+    // list of files / patterns to load in the browser
+    
+    files: [
```
Comment (nyambati) [inline karma.conf.js:22]: The `js` folder contains subfolders. What you have specified only picks files in the js folder but ignore the other folders `controllers` and `services`. Change your wildcards to cater for files the above named folders.
You have not also included the angular dependencies refer to this [article ](http://ericnish.io/blog/set-up-
jasmine-and-karma-for-angularjs/)and make the necessary changes.

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        'spec/*.js',
```
Comment (nyambati) [inline karma.conf.js]: Our Tests will reside in the test folder and therefore this should be changed to `test/client`. to separate backend and front-end tests.

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,14 @@ public/lib
 .sass-cache/
 .DS_store
 c4h_keys.txt
+<<<<<<< HEAD
```
Comment (nyambati) [inline .gitignore]: You failed to clean this file, make sure all the merge conflict diff are removed

**Review by nyambati — CHANGES_REQUESTED** [review]
Good Job 👍 . Clean up the git ignore and all should be good for the demo tomorrow.

File: .gitignore (hunk 3 above)
Comment (cnwosu) [reply .gitignore]: ok, thanks

**Review by nyambati — APPROVED** [review]

**Pull request merged**